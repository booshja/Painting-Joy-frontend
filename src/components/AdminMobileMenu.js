// dependencies
import React, { useContext } from "react";
import styled from "styled-components";
// components
import { NavLink } from "react-router-dom";
// context
import MenuContext from "../context/MenuContext";

const StyledMenu = styled.nav`
    z-index: 100;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%;
    width: 100%;
    text-align: center;
    position: absolute;
    top: 0;
    left: 0;
    transition: transform 0.3s ease-in-out;
    transform: ${({ open }) => (open ? "translateY(0)" : "translateY(-100%)")};
`;

const StyledLink = styled(NavLink)`
    font-family: "News Cycle", sans-serif;
    font-weight: 400;
    letter-spacing: 2px;
    font-size: 2rem;
    text-transform: uppercase;
    padding: 1rem 0;
    color: #207070;
    transition: color 0.3s linear;

    &.active {
        font-weight: 700;
    }
`;

const MenuWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 1rem;
`;

const AdminMobileMenu = () => {
    const { menuOpen, setMenuOpen } = useContext(MenuContext);

    return (
        <StyledMenu open={menuOpen}>
            <MenuWrapper>
                <StyledLink
                    exact
                    to="/admin/dashboard"
                    onClick={() => setMenuOpen(false)}
                >
                    Dashboard
                </StyledLink>
                <StyledLink
                    exact
                    to="/admin/murals"
                    onClick={() => setMenuOpen(false)}
                >
                    Murals
                </StyledLink>
                <StyledLink
                    exact
                    to="/admin/messages"
                    onClick={() => setMenuOpen(false)}
                >
                    Messages
                </StyledLink>
                <StyledLink
                    exact
                    to="/admin/items"
                    onClick={() => setMenuOpen(false)}
                >
                    Store Items
                </StyledLink>
                <StyledLink
                    exact
                    to="/admin/orders"
                    onClick={() => setMenuOpen(false)}
                >
                    Orders
                </StyledLink>
                <StyledLink
                    exact
                    to="/admin/reporting"
                    onClick={() => setMenuOpen(false)}
                >
                    Reporting
                </StyledLink>
                {/* <StyledLink
                    exact
                    to="/admin/ig-edit"
                    onClick={() => setMenuOpen(false)}
                >
                    Instagram Feed
                </StyledLink> */}
            </MenuWrapper>
        </StyledMenu>
    );
};

export default AdminMobileMenu;
