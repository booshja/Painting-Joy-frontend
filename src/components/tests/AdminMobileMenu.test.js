import { render } from "@testing-library/react";
import AdminMobileMenu from "../AdminMobileMenu";
import MenuContext from "../../context/MenuContext";
import { BrowserRouter } from "react-router-dom";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing, menuOpen true", () => {
    render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true, setMenuOpen: null }}>
                <AdminMobileMenu />
            </MenuContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, menuOpen false", () => {
    render(
        <BrowserRouter>
            <MenuContext.Provider
                value={{ menuOpen: false, setMenuOpen: null }}
            >
                <AdminMobileMenu />
            </MenuContext.Provider>
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot, menuOpen true", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true, setMenuOpen: null }}>
                <AdminMobileMenu />
            </MenuContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, menuOpen false", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <MenuContext.Provider
                value={{ menuOpen: false, setMenuOpen: null }}
            >
                <AdminMobileMenu />
            </MenuContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, menuOpen true", () => {
    const { getByText } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true, setMenuOpen: null }}>
                <AdminMobileMenu />
            </MenuContext.Provider>
        </BrowserRouter>
    );

    expect(getByText("Dashboard")).toBeInTheDocument();
    expect(getByText("Murals")).toBeInTheDocument();
    expect(getByText("Messages")).toBeInTheDocument();
    expect(getByText("Store Items")).toBeInTheDocument();
    expect(getByText("Orders")).toBeInTheDocument();
    expect(getByText("Log Out")).toBeInTheDocument();
});

it("renders correctly, menuOpen false", () => {
    const { getByText } = render(
        <BrowserRouter>
            <MenuContext.Provider
                value={{ menuOpen: false, setMenuOpen: null }}
            >
                <AdminMobileMenu />
            </MenuContext.Provider>
        </BrowserRouter>
    );

    expect(getByText("Dashboard")).toBeInTheDocument();
    expect(getByText("Murals")).toBeInTheDocument();
    expect(getByText("Messages")).toBeInTheDocument();
    expect(getByText("Store Items")).toBeInTheDocument();
    expect(getByText("Orders")).toBeInTheDocument();
    expect(getByText("Log Out")).toBeInTheDocument();
});
