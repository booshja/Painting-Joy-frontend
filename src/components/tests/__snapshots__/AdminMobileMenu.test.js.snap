// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, menuOpen false 1`] = `
<DocumentFragment>
  .c0 {
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c2 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c2.active {
  font-weight: 700;
}

.c3 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c3.active {
  font-weight: 700;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

@media screen and (min-width:1024px) {
  .c0 {
    display: none;
  }
}

<nav
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2"
        href="/admin/dashboard"
      >
        Dashboard
      </a>
      <a
        class="c2"
        href="/admin/murals"
      >
        Murals
      </a>
      <a
        class="c2"
        href="/admin/messages"
      >
        Messages
      </a>
      <a
        class="c2"
        href="/admin/items"
      >
        Store Items
      </a>
      <a
        class="c2"
        href="/admin/orders"
      >
        Orders
      </a>
      <a
        class="c3"
      >
        Log Out
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot, menuOpen true 1`] = `
<DocumentFragment>
  .c0 {
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c2 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c2.active {
  font-weight: 700;
}

.c3 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c3.active {
  font-weight: 700;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

@media screen and (min-width:1024px) {
  .c0 {
    display: none;
  }
}

<nav
    class="c0"
    open=""
  >
    <div
      class="c1"
    >
      <a
        class="c2"
        href="/admin/dashboard"
      >
        Dashboard
      </a>
      <a
        class="c2"
        href="/admin/murals"
      >
        Murals
      </a>
      <a
        class="c2"
        href="/admin/messages"
      >
        Messages
      </a>
      <a
        class="c2"
        href="/admin/items"
      >
        Store Items
      </a>
      <a
        class="c2"
        href="/admin/orders"
      >
        Orders
      </a>
      <a
        class="c3"
      >
        Log Out
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
